src/components/Utils/orderResource.js

```diff
@@ -57,5 +64,5 @@ export const cloneOrder = (order, mutator, lines) => {
     }));
   }
 
-  return saveOrder(clonedOrder, mutator);
+  return poNumber ? saveOrder(clonedOrder, mutator) : Promise.reject();
```

Comment:
this fn is written using asyn\await, IMO its better to use `throw` instead of `Promise`

Reply:
also throwing might be done earlier, near the place you've got poNumber